### ISSUE TITLE
introspection and nameField:true

### ISSUE THREAD

**Issue by sebastian-philipp** [django-salesforce/django-salesforce#308]
Hi folks,

I have contradicting `blank=True, null=True` values for our name field: 

![image (3)](https://user-images.githubusercontent.com/2574405/178217781-2cbe90f5-7877-4e10-883d-4a36f2cb61b3.png)

The field introspection looks like so: 

```python
    name = models.CharField(
        db_column='Name',
        max_length=80,
        verbose_name='Project Name',
        default=models.DEFAULTED_ON_CREATE,
        blank=True,
        null=True,
    )
```
and

```
{
			"aggregatable": true,
			"aiPredictionField": false,
			"autoNumber": false,
			"byteLength": 240,
			"calculated": false,
			"calculatedFormula": null,
			"cascadeDelete": false,
			"caseSensitive": false,
			"compoundFieldName": null,
			"controllerName": null,
			"createable": true,
			"custom": false,
			"defaultValue": null,
			"defaultValueFormula": null,
			"defaultedOnCreate": true,
			"dependentPicklist": false,
			"deprecatedAndHidden": false,
			"digits": 0,
			"displayLocationInDecimal": false,
			"encrypted": false,
			"externalId": false,
			"extraTypeInfo": null,
			"filterable": true,
			"filteredLookupInfo": null,
			"formulaTreatNullNumberAsZero": false,
			"groupable": true,
			"highScaleNumber": false,
			"htmlFormatted": false,
			"idLookup": true,
			"inlineHelpText": null,
			"label": "Project Name",
			"length": 80,
			"mask": null,
			"maskType": null,
			"name": "Name",
			"nameField": true,
			"namePointing": false,
			"nillable": true,
			"permissionable": false,
			"picklistValues": [],
			"polymorphicForeignKey": false,
			"precision": 0,
			"queryByDistance": false,
			"referenceTargetField": null,
			"referenceTo": [],
			"relationshipName": null,
			"relationshipOrder": null,
			"restrictedDelete": false,
			"restrictedPicklist": false,
			"scale": 0,
			"searchPrefilterable": false,
			"soapType": "xsd:string",
			"sortable": true,
			"type": "string",
			"unique": false,
			"updateable": true,
			"writeRequiresMasterRead": false
		},
```

Please note, `"nameField": true,`. Should we mark the field as `blank=False, null=False` in case of a `nameField`?

**Comment by sebastian-philipp** [comment]
thank you for looking into it.

**Comment by hynekcer** [comment]
It is now fixed for Django 5.0 by commits [c61f66b](https://github.com/django-salesforce/django-salesforce/commit/c61f66bf0180789d9ce6291222bb4891a3fb8c99) and [84e0186](https://github.com/django-salesforce/django-salesforce/commit/84e0186f779d27d5abbe432775f991c41015d42d).  
You can use parameters `... db_default=models.DEFAULTED_ON_CREATE, blank=True`
These parameters are now also exported by `inspectdb` with Django 5.0.

Database fields have a new parameter [db_default=...](https://docs.djangoproject.com/en/5.0/ref/models/fields/#db-default) in Django 5.0 that allows exactly what Salesforce does. Let the server to assign some useful value if nothing is specified. The most frequent fields are:
- name => a prefix or date + number
- owner => current user
- language => language of the current user etc.

---

When I used the following simple `admin.py` and called runserver then almost every object could be created by Django admin if it is writable, including many system objects:
``` python
from salesforce.testrunner.example.universal_admin import register_omitted_classes
from . import models

register_omitted_classes(models)
```

**Comment by hynekcer** [comment]
It is fixed also on PyPI in Django-salesforce 5.0.1 that will be released just now.
